### PR TITLE
Swipe detection demo

### PR DESCRIPTION
**Pull request by smm-ncl**
PR for running the Swipe Detection Demo with Bokeh

### CODE REVIEW

File: demos/swipe_detection/swipe_detection_network.py [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+                                          Sparse: PySparseModelBitAcc,
+                                          LIF: PyLifModelBitAcc,
+                                          PropheseeCamera: PyPropheseeCameraRawReaderModel})
+
+        # Compilation
+        if executable is None:
+            compiler = Compiler()
+            executable = compiler.compile(self.frame_input, run_cfg=run_cfg)
+        # Store the Lava network, only needed if network changes
+        #save(processes=[self.ff_inp,
```
Comment (PhilippPlank) [inline demos/swipe_detection/swipe_detection_network.py:107]: Could this code live in a method and be triggered using a parameter given at the instantiation of the class?
This parameter could for example be a path/filename to store the network in. If given, the network is stored, if not given, then not.

File: demos/swipe_detection/swipe_detection_network.py [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+        for i in range(self.scaled_height):
+            for j in range(self.scaled_width):
+                ks = self.weights_config["kernel_size"]
+                self.rec_weights_left[i, j - ks :j, i, j] = \
+                    self.weights_config["w_rec"]
+
+        self.rec_weights_left = csr_matrix(
+            self.rec_weights_left.reshape((self.num_neurons, self.num_neurons)))
+
+        # right weights
```
Comment (PhilippPlank) [inline demos/swipe_detection/swipe_detection_network.py:155]: capital letter

File: demos/swipe_detection/outprocess/process_out.py [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    def __init__(self, proc_params):
+        super().__init__(proc_params)
+        self.frame_shape = proc_params["frame_shape"]
+        self.direction_shape = proc_params["direction_shape"]
+        self._send_pipe = proc_params["send_pipe"]
+        self.height = self.frame_shape[2]
+        self.width = self.frame_shape[3]
+
+    def run_spk(self):
+
```
Comment (PhilippPlank) [inline demos/swipe_detection/outprocess/process_out.py:55]: remove blank line

File: demos/swipe_detection/swipe_detection_network.py [hunk 4]
```diff
@@ -0,0 +1,266 @@
...
+blocking_default_config = False
+
+
+class SwipeDetector:
+    """Class to setup the swipe detection network, compile it, and initialize
+         its runtime."""
+
+    def __init__(self,
+                 send_pipe: type(Pipe),
+                 num_steps: int,
```
Comment (PhilippPlank) [inline demos/swipe_detection/swipe_detection_network.py:68]: Docstring for the input parameters would be nice, since this is the main class

**Review by PhilippPlank — APPROVED** [review]
Looks good. I left some comments for minor changes, mainly documentation.

File: demos/swipe_detection/main_swipe_demo.py [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+    from bokeh.plotting import figure, curdoc
+    from bokeh.layouts import gridplot, Spacer
+    from bokeh.models import LinearColorMapper, ColorBar, Title, Button
+    from bokeh.models.ranges import DataRange1d
+except ModuleNotFoundError:
+    print("Module 'bokeh' is not installed. Please install module 'bokeh' in"
+          " order to run the motion tracking demo.")
+    exit()
+from swipe_detection_network import SwipeDetector
+from bokeh.models import Arrow, NormalHead
```
[the remark this answers is not in the document]
Comment (smm-ncl) [reply demos/swipe_detection/main_swipe_demo.py:22]: Good point

File: demos/swipe_detection/main_swipe_demo.py [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+# ==========================================================================
+recv_pipe, send_pipe = multiprocessing.Pipe()
+num_steps = 400
+
+# Checks whether terminate button has been clicked and allows to stop
+# updating the bokeh doc
+stop_button_pressed: bool = False
+use_loihi2 = Loihi2.is_loihi2_available
+
+_, executable = load("swipe_detector.pickle")
```
[the remark this answers is not in the document]
Comment (smm-ncl) [reply demos/swipe_detection/main_swipe_demo.py:37]: Done

File: demos/swipe_detection/swipe_detection_network.py [hunk 7]
```diff
@@ -0,0 +1,266 @@
...
+        self.runtime.initialize()
+
+    def _create_processes(self) -> None:
+        # Create Processes and Weights
+        self.frame_input = PropheseeCamera(**self.prophesee_input_config)
+        self.num_neurons = np.prod(self.frame_input.s_out.shape)
+        _, _, self.scaled_height, self.scaled_width = self.frame_input.shape
+        self.flat_shape = (self.num_neurons,)
+
+        # create weights
```
[the remark this answers is not in the document]
Comment (smm-ncl) [reply demos/swipe_detection/swipe_detection_network.py:136]: Done

File: demos/swipe_detection/swipe_detection_network.py (hunk 1 above)
Comment (smm-ncl) [reply demos/swipe_detection/swipe_detection_network.py:107]: Done

File: demos/swipe_detection/swipe_detection_network.py (hunk 2 above)
Comment (smm-ncl) [reply demos/swipe_detection/swipe_detection_network.py:155]: Done

File: demos/swipe_detection/outprocess/process_out.py (hunk 3 above)
Comment (smm-ncl) [reply demos/swipe_detection/outprocess/process_out.py:55]: Done

File: src/lava/lib/peripherals/dvs/prophesee.py [hunk 8]
```diff
@@ -54,29 +54,35 @@ class PropheseeCamera(AbstractProcess):
         Transformations to be applied to the events before sending them out.
     num_output_time_bins: int
         The number of output time bins to use for the ToFrame transformation.
+    sync_time: bool
+        Whether to
```
[the remark this answers is not in the document]
Comment (smm-ncl) [reply src/lava/lib/peripherals/dvs/prophesee.py]: Done

File: demos/swipe_detection/swipe_detection_network.py (hunk 4 above)
Comment (smm-ncl) [reply demos/swipe_detection/swipe_detection_network.py:97]: Done

**Review by gkarray — APPROVED** [review]

**Pull request closed without merging**